src/jarabe/desktop/schoolserver.py

```diff
@@ -122,44 +171,85 @@ def register_laptop(url=_REGISTER_URL):
         uuid_ = str(uuid.uuid1())
     sn = sn or 'SHF00000000'
     uuid_ = uuid_ or '00000000-0000-0000-0000-000000000000'
-
     nick = get_nick_name()
-
     settings = Gio.Settings('org.sugarlabs.collaboration')
     jabber_server = settings.get_string('jabber-server')
-    _store_identifiers(sn, uuid_, jabber_server)
 
+    # Check server for registration
     if jabber_server:
+        db_url = 'http://' + jabber_server + ':5000/'
         url = 'http://' + jabber_server + ':8080/'
-
-    if sys.hexversion < 0x2070000:
-        server = xmlrpclib.ServerProxy(url, _TimeoutTransport())
-    else:
-        socket.setdefaulttimeout(_REGISTER_TIMEOUT)
-        server = xmlrpclib.ServerProxy(url)
     try:
-        data = server.register(sn, nick, uuid_, profile.pubkey)
-    except (xmlrpclib.Error, TypeError, socket.error):
-        logging.exception('Registration: cannot connect to server')
-        raise RegisterError(_('Cannot connect to the server.'))
-    finally:
-        socket.setdefaulttimeout(None)
-
-    if data['success'] != 'OK':
-        logging.error('Registration: server could not complete request: %s',
-                      data['error'])
-        raise RegisterError(_('The server could not complete the request.'))
-
-    settings.set_string('jabber-server', data['jabberserver'])
+        response = urllib2.urlopen(db_url)
```

Comment:
Please use libsoup for opening urls.  Libsoup properly intergrates with the gobject mainloop; so that the ui is still responsive while we wait on the network.

Lib soup has documentation, but you can probs just copy some code from my reddit client;  https://github.com/samdroid-apps/something-for-reddit/blob/master/src/api.py#L182